home/rvagg/2021-02-exploration-report-small-filecoin-deals.md

```diff
@@ -0,0 +1,158 @@
+  * The miner deciding whether to accept the funds. Miners set default deal parameters but can set additional configuration options and even insert an "deal acceptance script" to their deal flow to accept and reject deals based on all of the parameters. See https://docs.filecoin.io/mine/lotus/miner-configuration/#using-filters-for-fine-grained-storage-and-retrieval-deal-acceptance
+    * This can be slow too, I had deals stuck in `StorageDealCheckForAcceptance` for extended periods, I'm not sure what this is but maybe it's waiting for manual acceptance by the miner?
+  * Transferring the data via graphsync to the miner
+  * The miner setting up the deal with the chain (does this come before or after the next step?)
+  * The miner queueing the piece for sealing (can be slow, miners can choose to batch deals, which defer deals until enough have been selected, and can set maximum number of deals per sector too)
+  * Sealing the sector that includes the piece (slow)
+  * 12-24h for the entire process is apparently a reasonable amount of time (slow)
+* Successfully made a deal with **only one miner**, [`f022352`](https://filfox.info/en/address/f022352) in Norway! As of writing, still waiting for PreCommit, ~5 hours later.
+
+### Filecoin Plus (FIL+) deals
+
+* FIL+ details are at https://github.com/filecoin-project/filecoin-plus-client-onboarding/ - information about notaries and clients asking for DataCap for making FIL+ deals
+* https://verify.glif.io/ will give you 8 GiB of DataCap every month if you have an established GitHub account, without going through a Notary.
+* https://verify.glif.io/ will also let you look at DataCap balance for a wallet/ID.
+* `lotus-shed` (comes with the Lotus source, `make lotus-shed`) has a `verifreg` command that's being built out that you can use to check a client's balance `lotus-shed verifreg check-client <address/ID>`
+  * `lotus-shed` has other interesting commands where you can look up notaries and see all of the clients with DataCap.
+* I got some DataCap to try it out through glif.
+* Using `lotus client deal` to attempt some deals again, I was able to say `yes` to "verified deal", this yields new prices, some of which are `0 FIL`:
+  - f019104 0 FIL $0
+  - f019041 ~0.000000405 FIL (0.00000000000078125 FIL per epoch)
+  - f066596 0 FIL $0
+  - f03134 ~8100 FIL (0.015625 FIL per epoch) ~$283,500 _(obviously I said `no` to this one)_
+  - f023565 0 FIL $0
+  - f022352 0 FIL $0
+  - f071624 ~0.000000405 FIL (0.00000000000078125 FIL per epoch)
+  - f0135078 ~0.000000405 FIL (0.00000000000078125 FIL per epoch)
+  - f058369 ~0.00081 FIL (0.0000000015625 FIL per epoch)
+  - f010479 ~0 FIL (0 FIL per epoch)
+  - f01247 ~0 FIL (0 FIL per epoch)
+* Successfully made a deal with the same miner as for my unverified deal, [`f022352`](https://filfox.info/en/address/f022352). As of writing, still waiting for PreCommit, ~5 hours later. 0 FIL instead of 0.0001458 FIL.
```

Comment:
Interesting, does the retrieval cost change at all between these two deals?

Reply:
...and also whether you are actually able to retrieve your data from the miner.

Reply:
... no ... I haven't poked at retrieval at all in this process so I'm pretty ignorant, and I don't expect that to be straightforward, which is why I've stopped sort to focus on the _getting things in_ question.